run_tests.py

```diff
@@ -0,0 +1,313 @@
+        # application bundle in /Contents/MacOS/Sketch and the bundle also
+        # includes binaries for XPC services such as Mirror and Assistants.
+        # Furthermore, pre-release builds of Sketch have their binaries named
+        # differently, e.g. Sketch Beta.
+        if exe and exe.startswith(f"{path}/Contents/MacOS"):
+            print(f"Terminating process: {proc.info}", file=sys.stderr)
+            proc.kill()
+            break
+
+
+def main(argv):
+    sketch = '/Applications/Sketch.app'  # default Sketch installation path
+    plugin = ''
+    output_file_path = ''
+    timeout = 30
+
+    usage = 'run_tests.py -s <sketch> -p <plugin> -o <outputFilePath> [-t <timeout>]'
+    try:
+        opts, args = getopt.getopt(
+            argv, "hs:p:o:t:", [
+                "sketch=", "plugin=", "outputFilePath=", "timeout="])
+    except getopt.GetoptError:
+        print(usage)
+        sys.exit(2)
+
+    for opt, arg in opts:
+        if opt == '-h':
+            print(usage)
+            sys.exit()
+        elif opt in ("-s", "--sketch"):
+            sketch = Path(arg).expanduser().resolve()
+        elif opt in ("-p", "--plugin"):
+            plugin = Path(arg).expanduser().resolve()
+        elif opt in ("-o", "--outputFilePath"):
+            output_file_path = Path(arg).expanduser().resolve()
+        elif opt in ("-t", "--timeout"):
+            timeout = float(arg)
+
+    if not plugin or not output_file_path:
+        print(usage)
+        sys.exit(2)
+
+    if not os.path.exists(plugin):
+        print(f"Plugin not found at: {plugin}", file=sys.stderr)
+        sys.exit(2)
+    
+    if not os.path.exists(sketch):
+        print(f"Sketch bundle not found at: {sketch}", file=sys.stderr)
+        sys.exit(2)
+    
+    # create a symbolic link to the plugin because Sketch expects it to
+    # be inside the Application Support Plugins folder.
+    plugin_path = PurePath(
+        str(Path.home()),
+        "Library/Application Support/com.bohemiancoding.sketch3/Plugins",
+        os.path.basename(plugin))
+
+    # in cases when the execution is stopped, make sure to remove the symbolic
+    # link before adding the new one
+    # We need the two tests because a broken symlink returns false, but the symlink
+    # is still actually there. If it's still actually there, the create symlink command barfs.
+    if os.path.exists(plugin_path) or os.path.islink(plugin_path):
+        os.remove(plugin_path)
+
+    os.symlink(os.path.abspath(plugin), plugin_path, target_is_directory=True)
+
+    # remove any previous test results
+    if os.path.exists(output_file_path):
+        os.remove(output_file_path)
+
+    # start execution time
+    start_time = time.time()
+
+    # get the plugin identifier from the manifest, Sketch uses the
+    # identifier to look up the corresponding plugin.
+    with open(PurePath(plugin_path, 'Contents/Sketch/manifest.json'), 'r') as f:
+        data = f.read()
```

Comment:
Here's another spot where you can combine the two. Just do json.load(f) and delete the f.read()